File: javascript/leetcode/49.js
const strs = ["eat", "tea", "tan", "ate", "nat", "bat"];

var groupAnagrams = function (strs) {
  const obj = {};
  for (let str of strs) {
    const sorted_str = str.split("").sort().join("");
    if (sorted_str in obj) {
      obj[sorted_str].push(str);
    } else {
      obj[sorted_str] = [str];
    }
  }

  const answer = [];
  for (const [key, value] of Object.entries(obj)) {
    answer.push(value);
  }

  return answer;
};

groupAnagrams(strs);

File: python/leetcode/49.py
strs = ["eat", "tea", "tan", "ate", "nat", "bat"]


def groupAnagrams(strs):
    dic = {}
    for str in strs:
        sorted_str = "".join(sorted(str))

        if sorted_str in dic:
            dic[sorted_str].append(str)
        else:
            dic[sorted_str] = [str]

    answer = []

    for value in dic.values():
        answer.append(value)

    return answer


groupAnagrams(strs)
